src/custom-types.d.ts

```diff
@@ -13,8 +13,23 @@ interface HTMLElement {
 
 // This lint rule does not apply to type definitions:
 // eslint-disable-next-line no-unused-vars
-type ViewPartial<ViewPartialParams = object> = (data: ViewPartialParams) => string;
-type ViewPartialData<ViewPartialParams = object> = {
-    partial: ViewPartial<ViewPartialParams>,
-    nonce: string
+type ViewPartial<ViewPartialParams = object> = (data: ViewPartialParams & { partial: { name: string } }) => string;
+type GuestViewPartialData<ViewPartialParams = object> = {
+    partial: ViewPartial<ViewPartialParams>;
+    nonce: string;
   } & ViewPartialParams;
+type MainViewPartialData<ViewPartialParams = object> = {
+    fxaProfile: NonNullable<import('express').Request['user']>['fxa_profile_json'];
+  } & GuestViewPartialData<ViewPartialParams>;
+
+declare namespace Express {
+  export interface Request {
+    user?: {
+      // TODO: Finish the type definition of the user object
```

Comment:
Just FYI: I believe this is the schema for the fxa response: https://github.com/mozilla/fxa/blob/main/packages/fxa-profile-server/lib/routes/profile.js#L63-L77.

Reply:
Ah thanks, added in 0045fdf48.